Source/JavaScriptCore/parser/ASTBuilder.h

```diff
@@ -980,7 +980,7 @@ class ASTBuilder {
     ExpressionNode* createAssignment(const JSTokenLocation& location, int& assignmentStackDepth, ExpressionNode* rhs, int initialAssignmentCount, int currentAssignmentCount, const JSTextPosition& lastTokenEnd)
     {
         AssignmentInfo& info = m_assignmentInfoStack.last();
-        ExpressionNode* result = makeAssignNode(location, info.m_node, info.m_op, rhs, info.m_initAssignments != initialAssignmentCount, info.m_initAssignments != currentAssignmentCount, info.m_start, info.m_divot + 1, lastTokenEnd);
+        ExpressionNode* result = makeAssignNode(location, info.m_node, info.m_op, rhs, info.m_initAssignments != initialAssignmentCount, info.m_initAssignments + 1 != currentAssignmentCount, info.m_start, info.m_divot + 1, lastTokenEnd);
         m_assignmentInfoStack.removeLast();
         assignmentStackDepth--;
         return result;
```

Comment:
Can you ensure that this function is only called in assignment context? Is it possible that it is called in a bit complex situation (e.g. for-in's variable) and they are missing assignment counting (since syntax does not have = etc.)?

Reply:
Thanks, I checked and `makeAssignNode` is only called from here and `createAssignment` is also only called from one place.